pkg/messaging/mqtt/setup_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package mqtt_test
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	"github.com/mainflux/mainflux/pkg/messaging/mqtt"
+	dockertest "github.com/ory/dockertest/v3"
+)
+
+var (
+	publisher messaging.Publisher
+	pubsub    messaging.PubSub
+)
+
+func TestMain(m *testing.M) {
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+
+	container, err := pool.Run("vernemq/vernemq", "latest", []string{})
+	if err != nil {
+		log.Fatalf("Could not start container: %s", err)
+	}
+	handleInterrupt(pool, container)
+
+	address := fmt.Sprintf("%s:%s", "localhost", container.GetPort("1883"))
+	if err := pool.Retry(func() error {
+		publisher, err = mqtt.NewPublisher(address, 30*time.Second)
+		return err
+	}); err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
```

Comment:
@drasko 
Can you please review this part?
It seems correct from my side, but the tests are failing to start: `could not connect to docker: reached retry deadline`